web/src/views/User.vue

```diff
@@ -27,48 +29,61 @@
   </FluidContainer>
 </template>
 
-<script lang="ts">
-import { computed, defineComponent, onMounted, ref } from 'vue';
+<script lang="ts" setup>
+import { useLocalStorage } from '@vueuse/core';
+import dayjs from 'dayjs';
+import TimeAgo from 'javascript-time-ago';
+import { computed, onMounted, ref } from 'vue';
 import { useI18n } from 'vue-i18n';
 
 import Button from '~/components/atomic/Button.vue';
+import SelectField from '~/components/form/SelectField.vue';
 import FluidContainer from '~/components/layout/FluidContainer.vue';
 import useApiClient from '~/compositions/useApiClient';
 
-export default defineComponent({
-  name: 'User',
+const { t, availableLocales, locale } = useI18n();
 
-  components: {
-    FluidContainer,
-    Button,
-  },
+const apiClient = useApiClient();
+const token = ref<string | undefined>();
 
-  setup() {
-    const apiClient = useApiClient();
-    const token = ref<string | undefined>();
+onMounted(async () => {
+  token.value = await apiClient.getToken();
+});
 
-    onMounted(async () => {
-      token.value = await apiClient.getToken();
-    });
+// eslint-disable-next-line no-restricted-globals
+const address = `${location.protocol}//${location.host}`;
 
-    // eslint-disable-next-line no-restricted-globals
-    const address = `${location.protocol}//${location.host}`;
+const usageWithShell = computed(() => {
+  let usage = `export WOODPECKER_SERVER="${address}"\n`;
+  usage += `export WOODPECKER_TOKEN="${token.value}"\n`;
+  return usage;
+});
 
-    const usageWithShell = computed(() => {
-      let usage = `export WOODPECKER_SERVER="${address}"\n`;
-      usage += `export WOODPECKER_TOKEN="${token.value}"\n`;
-      return usage;
-    });
+const usageWithCurl = `# ${t(
+  'user.shell_setup_before',
+)}\ncurl -i \${WOODPECKER_SERVER}/api/user -H "Authorization: Bearer \${WOODPECKER_TOKEN}"`;
 
-    const usageWithCurl = `# ${useI18n().t(
-      'user.shell_setup_before',
-    )}\ncurl -i \${WOODPECKER_SERVER}/api/user -H "Authorization: Bearer \${WOODPECKER_TOKEN}"`;
+const usageWithCli = `# ${t('user.shell_setup_before')}\nwoodpecker info`;
 
-    const usageWithCli = `# ${useI18n().t('user.shell_setup_before')}\nwoodpecker info`;
+const cliDownload = 'https://github.com/woodpecker-ci/woodpecker/releases';
 
-    const cliDownload = 'https://github.com/woodpecker-ci/woodpecker/releases';
+const localeOptions = computed(() =>
+  availableLocales.map((availableLocale) => ({
+    value: availableLocale,
+    text: new Intl.DisplayNames(availableLocale, { type: 'language' }).of(availableLocale) || availableLocale,
+  })),
+);
 
-    return { token, usageWithShell, usageWithCurl, usageWithCli, cliDownload, address };
+const storedLocale = useLocalStorage('woodpecker:locale', locale.value);
```

Comment:
we should come up with naming for local/session storage names.
Currently we have `last-badge-type`, `log-auto-scroll` (without `woodpecker:` prefix)

Reply:
I would prefer a prefix, mostly to avoid conflict when developing on localhost.

Reply:
is this resolved ?!?

Reply:
to have a single source of truth why not add the save func to web/src/utils/locale.ts ?

Reply:
I agree the locale configs are a bit all over the place, but I'm not really sure how to improve this at the moment, as it is not a simple save function.

I think we should tackle this after #1345.